recipes/server_check_helper.rb

```diff
@@ -0,0 +1,30 @@
+require 'ipaddr'
+
+def is_server_allowed?(retry_join_list, my_ip)
+  if retry_join_list.nil? || retry_join_list.empty?()
+    return true
```

Comment:
why return true here?  should we disallow a node to become a server if the retry_join is invalid?

Reply:
Currently because we use Consul to block the server from starting,  so we probably only want to do this if there is an explicit retry list specified,  otherwise every server in that datacenter will be blocked.

Reply:
This recipe is only enabled when `node['consul']['config']['server'] === true`, right?
If a server has an invalid retry_join list, we should disable the server.
The retry_join list can be a DNS name, for example: `[ "consul.service.<dc>.consul"]`.  If a node has this config, it should not be allowed to be a consul server.

Reply:
> Allow cases a cluster that maybe only has a single node and don't have a retry_join list defined yet.

We don't do this anywhere. Always 5 or 3 servers.

> There are cases where the consul-cluster role itself has a dynamic retry_join list, that we do not yet understand how to parse.

This is handled in the conditional below with `is_valid_ip_address(x)?`, right? If not a list of IPs, it'll default to `true` at the end.

I'm on the fence, but side with @mechpen on this one. Perhaps an empty or missing retry_join should be false.

Reply:
> > Allow cases a cluster that maybe only has a single node and don't have a retry_join list defined yet.
> 
> We don't do this anywhere. Always 5 or 3 servers.  
> 
> > There are cases where the consul-cluster role itself has a dynamic retry_join list, that we do not yet understand how to parse.
> 
> This is handled in the conditional below with `is_valid_ip_address(x)?`, right? If not a list of IPs, it'll default to `true` at the end.
> 
> I'm on the fence, but side with @mechpen on this one. Perhaps an empty or missing retry_join should be false.

I see, what I meant was that currently the logic is to default to allow in these cases:
1. nil or empty retry_join   ( I will change this to disallow per comments).
2. dynamic list (will keep defaulting to allow).